targets/dummy/dummynode.cpp

```diff
@@ -308,20 +312,47 @@ extern "C" void AsebaAssert(AsebaVMState *vm, AsebaAssertReason reason)
 	AsebaVMInit(vm);
 }
 
+int usage(char* program)
+{
+	std::cerr << "Usage: " << program << " [--port|-p PORT] [ID, from 0 to 9]" << std::endl;
+	std::cerr << "Usage: " << program << " --help|-h" << std::endl;
+	std::cerr << "Creates one node dummynode-ID with node id ID+1 listening on port:" << std::endl;
+	std::cerr << " - a dynamically chosen port, if PORT == 0" << std::endl;
+	std::cerr << " - PORT, if PORT != 0 and PORT is available" << std::endl;
+	std::cerr << " - 33333+ID, if PORT is not set and 33333+ID is available." << std::endl;
+	std::cerr << "The Dashel target is printed on stdout." << std::endl;
+	return 1;
+}
 
 int main(int argc, char* argv[])
 {
-	const int basePort = ASEBA_DEFAULT_PORT;
-	int deltaPort = 0;
-	if (argc > 1)
+	int port(ASEBA_DEFAULT_PORT);
+	bool do_delta(true);
+	int deltaNodeId(0);
+
+	int argCounter = 1;
+	while (argCounter < argc)
 	{
-		deltaPort = atoi(argv[1]);
-		if (deltaPort < 0 || deltaPort >= 9)
+		const char *arg = argv[argCounter++];
+		if ((strcmp(arg, "-p") == 0) || (strcmp(arg, "--port") == 0))
+			do_delta = false, port = atoi(argv[argCounter++]);
+		else if ((strcmp(arg, "-h") == 0) || (strcmp(arg, "--help") == 0))
+			return usage(argv[0]);
+		else
 		{
-			std::cerr << "Usage: " << argv[0] << " [delta port, from 0 to 9]" << std::endl;
-			return 1;
+			deltaNodeId = atoi(arg);
+			if (deltaNodeId < 0 || deltaNodeId > 9)
+				return usage(argv[0]);
 		}
 	}
-	node.listen(basePort, deltaPort);
+
+	Dashel::Stream* listen = node.listen(do_delta ? port+deltaNodeId : port, deltaNodeId);
+
+	Aseba::Zeroconf zs;
+	std::string txt_record = Aseba::Zeroconf::txtRecord(ASEBA_PROTOCOL_VERSION, "Unknown", {1+deltaNodeId}, {ASEBA_PID_UNDEFINED});
```

Comment:
On the one hand, the name could be dummy rather than unknown; on the other hand, its pid is undefined...